flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/StructuredFunctionsITCase.java

```diff
@@ -197,6 +202,101 @@ private static Stream<TestSetSpec> objectOfTestCases() {
+                                                "n2",
+                                                call("Type2Constructor", 15, "Alice"))
+                                        .objectUpdate(
+                                                "n1",
+                                                objectOf(Type1.class, "a", 16, "b", "UpdatedBob")),
+                                "OBJECT_UPDATE(OBJECT_OF('"
+                                        + NestedType.class.getName()
+                                        + "', 'n1', Type1Constructor(f0, f1), 'n2', Type2Constructor(15, 'Alice')), "
+                                        + "'n1', OBJECT_OF('"
+                                        + Type1.class.getName()
+                                        + "', 'a', 16, 'b', 'UpdatedBob'))",
+                                NestedType.of(Type1.of(16, "UpdatedBob"), Type2.of(15, "Alice")),
+                                DataTypes.STRUCTURED(
+                                        NestedType.class.getName(),
+                                        DataTypes.FIELD(
+                                                "n1",
+                                                DataTypes.STRUCTURED(
+                                                        Type1.class.getName(),
+                                                        DataTypes.FIELD(
+                                                                "a", DataTypes.INT().notNull()),
+                                                        DataTypes.FIELD(
+                                                                "b",
+                                                                DataTypes.CHAR(10).notNull()))),
+                                        DataTypes.FIELD(
+                                                "n2",
+                                                DataTypes.STRUCTURED(
+                                                        Type2.class.getName(),
+                                                        DataTypes.FIELD("a", DataTypes.INT()),
+                                                        DataTypes.FIELD("b", DataTypes.STRING())))))
+                        // Test when class not found
+                        .testSqlResult(
+                                "OBJECT_UPDATE(OBJECT_OF('not.existing.clazz', 'a', 42, 'b', 'Bob'), 'b', 'Alice')",
+                                Row.of(42, "Alice"),
+                                DataTypes.STRUCTURED(
+                                        "not.existing.clazz",
+                                        DataTypes.FIELD("a", DataTypes.INT().notNull()),
+                                        DataTypes.FIELD("b", DataTypes.CHAR(5).notNull())))
+                        // Test update field to null
+                        .testResult(
+                                objectOf(Type1.class, "a", 42, "b", "Bob")
```

Comment:
add a test where the object itself is null, e.g. `OBJECT_UPDATED(CAST(NULL AS STRUCTURED<>), ...)`